server/svix-server/src/core/operational_webhooks.rs

```diff
@@ -0,0 +1,134 @@
+// SPDX-FileCopyrightText: © 2022 Svix Authors
+// SPDX-License-Identifier: MIT
+
+//! Module defining an interface for sending webhook events about the service.
+
+use chrono::{DateTime, Utc};
+use http::StatusCode;
+use serde::{Deserialize, Serialize};
+use svix::api::{MessageIn, Svix, SvixOptions};
+
+use super::{
+    security::{generate_management_token, Keys},
+    types::{
+        ApplicationId, ApplicationUid, EndpointId, EndpointUid, MessageAttemptId, MessageId,
+        OrganizationId,
+    },
+};
+use crate::error::{HttpError, Result};
+
+/// Sent when an endpoint has been automatically disabled after continuous failures.
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct EndpointDisabledEvent {
+    pub app_id: ApplicationId,
+    pub app_uid: Option<ApplicationUid>,
+    pub endpoint_id: EndpointId,
+    pub endpoint_uid: Option<EndpointUid>,
+    pub fail_since: DateTime<Utc>,
+}
+
+/// Sent when an endpoint is created, updated, or deleted
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct EndpointEvent {
+    pub app_id: ApplicationId,
+    pub app_uid: Option<ApplicationUid>,
+    pub endpoint_id: EndpointId,
+    pub endpoint_uid: Option<EndpointUid>,
+}
+
+/// Sent when a message delivery has failed (all of the retry attempts have been exhausted) as a
+/// "message.attempt.exhausted" type or after it's failed four times as a "message.attempt.failing"
+/// event.
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct MessageAttemptEvent {
+    pub app_id: ApplicationId,
+    pub app_uid: Option<ApplicationUid>,
+    pub msg_id: MessageId,
+    pub msg_event_id: Option<MessageAttemptId>,
+    pub endpoint_id: EndpointId,
+    pub last_attempt: DateTime<Utc>,
```

Comment:
Make all of them use lifetimes so that we don't need to clone when creating them.